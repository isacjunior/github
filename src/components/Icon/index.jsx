import React from 'react'
import PropTypes from 'prop-types'

const Icon = ({ icon }) => <i className={`fa ${icon}`}></i>

Icon.propTypes = {
  icon: PropTypes.string.isRequired
}

export default Icon