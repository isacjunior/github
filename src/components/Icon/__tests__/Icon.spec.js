import React from 'react'
import { mount } from 'enzyme'
import Icon from '../index'

describe('Icon', () => {
  it('Icon render', () => {
    mount(<Icon icon="fa-fa" />)
  })

  it('Snapshot Icon', () => {
    const wrapper = mount(<Icon icon="fa-fa" />)
    expect(wrapper).toMatchSnapshot()
  })
})
