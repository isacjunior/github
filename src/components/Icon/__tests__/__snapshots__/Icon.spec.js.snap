// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Snapshot Icon 1`] = `
<Icon
  icon="fa-fa"
>
  <i
    className="fa fa-fa"
  />
</Icon>
`;
